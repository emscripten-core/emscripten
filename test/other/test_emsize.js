var c;c||=typeof Module != 'undefined' ? Module : {};var aa="object"==typeof window,h="function"==typeof importScripts,l="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node&&"renderer"!=process.type,p=Object.assign({},c),r=(a,b)=>{throw b;},t="",u,v;
if(l){var fs=require("fs"),w=require("path");t=__dirname+"/";v=a=>{a=x(a)?new URL(a):w.normalize(a);return fs.readFileSync(a)};u=a=>{a=x(a)?new URL(a):w.normalize(a);return new Promise((b,d)=>{fs.readFile(a,void 0,(e,n)=>{e?d(e):b(n.buffer)})})};process.argv.slice(2);"undefined"!=typeof module&&(module.exports=c);r=(a,b)=>{process.exitCode=a;throw b;}}else if(aa||h)h?t=self.location.href:"undefined"!=typeof document&&document.currentScript&&(t=document.currentScript.src),t=t.startsWith("blob:")?"":
t.substr(0,t.replace(/[?#].*/,"").lastIndexOf("/")+1),h&&(v=a=>{var b=new XMLHttpRequest;b.open("GET",a,!1);b.responseType="arraybuffer";b.send(null);return new Uint8Array(b.response)}),u=a=>x(a)?new Promise((b,d)=>{var e=new XMLHttpRequest;e.open("GET",a,!0);e.responseType="arraybuffer";e.onload=()=>{200==e.status||0==e.status&&e.response?b(e.response):d(e.status)};e.onerror=d;e.send(null)}):fetch(a,{credentials:"same-origin"}).then(b=>b.ok?b.arrayBuffer():Promise.reject(Error(b.status+" : "+b.url)));
var ba=c.print||console.log.bind(console),y=c.printErr||console.error.bind(console);Object.assign(c,p);p=null;var z=c.wasmBinary,A,B=!1,C,D,E=[],F=[],ca=[],J=[];function da(){var a=c.preRun;a&&("function"==typeof a&&(a=[a]),a.forEach(ea));K(E)}function ea(a){E.unshift(a)}function fa(a){J.unshift(a)}var L=0,M=null,N=null,O=a=>a.startsWith("data:application/octet-stream;base64,"),x=a=>a.startsWith("file://"),P;
function Q(a){if(a==P&&z)return new Uint8Array(z);if(v)return v(a);throw"both async and sync fetching of the wasm failed";}function ha(a){return z?Promise.resolve().then(()=>Q(a)):u(a).then(b=>new Uint8Array(b),()=>Q(a))}function R(a,b,d){return ha(a).then(e=>WebAssembly.instantiate(e,b)).then(d,e=>{y(`failed to asynchronously prepare wasm: ${e}`);c.onAbort?.(e);e="Aborted("+e+")";y(e);B=!0;throw new WebAssembly.RuntimeError(e+". Build with -sASSERTIONS for more info.");})}
function ia(a,b){var d=P;z||"function"!=typeof WebAssembly.instantiateStreaming||O(d)||x(d)||l||"function"!=typeof fetch?R(d,a,b):fetch(d,{credentials:"same-origin"}).then(e=>WebAssembly.instantiateStreaming(e,a).then(b,function(n){y(`wasm streaming compile failed: ${n}`);y("falling back to ArrayBuffer instantiation");return R(d,a,b)}))}function S(a){this.name="ExitStatus";this.message=`Program terminated with exit(${a})`;this.status=a}
var K=a=>{a.forEach(b=>b(c))},ja=c.noExitRuntime||!0,ka=[null,[],[]],T="undefined"!=typeof TextDecoder?new TextDecoder:void 0,la={a:(a,b,d,e)=>{for(var n=0,W=0;W<d;W++){var oa=D[b>>2],X=D[b+4>>2];b+=8;for(var G=0;G<X;G++){var f=C[oa+G],H=ka[a];if(0===f||10===f){f=H;for(var k=0,m=k+NaN,q=k;f[q]&&!(q>=m);)++q;if(16<q-k&&f.buffer&&T)f=T.decode(f.subarray(k,q));else{for(m="";k<q;){var g=f[k++];if(g&128){var I=f[k++]&63;if(192==(g&224))m+=String.fromCharCode((g&31)<<6|I);else{var Y=f[k++]&63;g=224==(g&
240)?(g&15)<<12|I<<6|Y:(g&7)<<18|I<<12|Y<<6|f[k++]&63;65536>g?m+=String.fromCharCode(g):(g-=65536,m+=String.fromCharCode(55296|g>>10,56320|g&1023))}}else m+=String.fromCharCode(g)}f=m}(1===a?ba:y)(f);H.length=0}else H.push(f)}n+=X}D[e>>2]=n;return 0}},U=function(){function a(d){U=d.exports;A=U.b;d=A.buffer;c.HEAP8=new Int8Array(d);c.HEAP16=new Int16Array(d);c.HEAPU8=C=new Uint8Array(d);c.HEAPU16=new Uint16Array(d);c.HEAP32=new Int32Array(d);c.HEAPU32=D=new Uint32Array(d);c.HEAPF32=new Float32Array(d);
c.HEAPF64=new Float64Array(d);F.unshift(U.c);L--;c.monitorRunDependencies?.(L);0==L&&(null!==M&&(clearInterval(M),M=null),N&&(d=N,N=null,d()));return U}var b={a:la};L++;c.monitorRunDependencies?.(L);if(c.instantiateWasm)try{return c.instantiateWasm(b,a)}catch(d){return y(`Module.instantiateWasm callback failed with error: ${d}`),!1}P??=O("test_emsize.wasm")?"test_emsize.wasm":c.locateFile?c.locateFile("test_emsize.wasm",t):t+"test_emsize.wasm";ia(b,function(d){a(d.instance)});return{}}(),V=c._main=
(a,b)=>(V=c._main=U.d)(a,b),Z,ma;N=function na(){Z||pa();Z||(N=na)};function pa(){function a(){if(!Z&&(Z=1,c.calledRun=1,!B)){K(F);K(ca);c.onRuntimeInitialized?.();if(qa){var b=V;try{var d=b(0,0);ja||(c.onExit?.(d),B=!0);r(d,new S(d))}catch(e){e instanceof S||"unwind"==e||r(1,e)}}if(b=c.postRun)"function"==typeof b&&(b=[b]),b.forEach(fa);K(J)}}if(!(0<L)){if(!ma&&(ma=1,da(),0<L))return;c.setStatus?(c.setStatus("Running..."),setTimeout(()=>{setTimeout(()=>c.setStatus(""),1);a()},1)):a()}}
if(c.preInit)for("function"==typeof c.preInit&&(c.preInit=[c.preInit]);0<c.preInit.length;)c.preInit.pop()();var qa=!0;c.noInitialRun&&(qa=!1);pa();
