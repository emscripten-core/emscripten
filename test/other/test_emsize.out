section      size   addr
TYPE           41     10
IMPORT          7     53
FUNCTION        9     62
TABLE           5     73
MEMORY          6     80
GLOBAL          8     88
EXPORT         17     98
ELEM            9    117
CODE         1083    129
DATA           72   1214
JS		4799	0
Total		6056
